apps/nova/src/pages/UpcomingLessons.tsx

```diff
@@ -21,10 +54,37 @@ const UpcomingLessons: React.FC = () => {
     canceled: true,
   });
 
+  // useMutation hooks
+  const { mutate: createRating } = useCreateRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("createRating failed!"),
+  });
+
+  const { mutate: editRating } = useEditRatingTutor({
+    onSuccess: () => setRateDialogInfo((prev) => ({ ...prev, enabled: false })),
+    onError: () => console.log("editRating failed!"),
+  });
+
+  // useEffect hooks
   useEffect(() => {
     if (!user) return navigate(Route.Root);
   }, [navigate, user]);
 
+  useEffect(() => {
+    if (user?.role !== "student") return;
+    const tutorId = Number(params.get("rate-tutor-id"));
+    const lessonId = Number(params.get("rate-lesson-id"));
+    const enabled = tutorId > 0 && lessonId > 0;
+    setRateDialogInfo((prev) => ({ ...prev, tutorId, lessonId, enabled }));
+  }, [params, user?.role]);
+
+  useEffect(() => {
+    const rateId = ratingQuery.data?.list[0].id || null;
+    const rateValue = ratingQuery.data?.list[0].value || null;
+    const rateFeedback = ratingQuery.data?.list[0].feedback || null;
+    setRateDialogInfo((prev) => ({ ...prev, rateValue, rateFeedback, rateId }));
+  }, [ratingQuery.data]);
```

Comment:
What you are trying to achieve is not stated in the task description. **You should enable the student from rating the tutor only in case the student didn't rate the tutor yet.**

The logic here enable the student from updating his rating as well which is not mentioned in the specs.